Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1444,7 +1444,19 @@ private string GetAudioArguments(StreamState state)
             {
                 if (EncodingHelper.IsCopyCodec(audioCodec))
                 {
-                    return "-acodec copy -strict -2";
+                    var segmentFormat = HlsHelpers.GetSegmentFileExtension(state.Request.SegmentContainer).TrimStart('.');
+                    var bitStreamArgs = string.Empty;
+
+                    // Apply aac_adtstoasc bitstream filter when media source is in mpegts.
+                    if (string.Equals(segmentFormat, "mp4", StringComparison.OrdinalIgnoreCase)
+                        && (string.Equals(state.MediaSource.Container, "mpegts", StringComparison.OrdinalIgnoreCase)
+                            || string.Equals(state.MediaSource.Container, "hls", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        bitStreamArgs = _encodingHelper.GetBitStreamArgs(state.AudioStream);
+                        bitStreamArgs = !string.IsNullOrEmpty(bitStreamArgs) ? " " + bitStreamArgs : string.Empty;
```

Comment:
slight optimization.
```suggestion
                        bitStreamArgs = string.IsNullOrEmpty(bitStreamArgs) ? string.Empty : " " + bitStreamArgs;
```